src/code/code_800EC960.c

```diff
@@ -4960,35 +4973,55 @@ void func_800F64E0(u8 arg0) {
     }
 }
 
-void func_800F6584(u8 arg0) {
+/**
+ * Enable or disable Malon's singing
+ *
+ * @param malonSingingDisabled true to disable, false to enable
+ */
+void Audio_ToggleMalonSinging(u8 malonSingingDisabled) {
     u8 seqPlayerIndex;
-    u16 sp34;
+    u16 channelMaskDisable;
+
+    sMalonSingingDisabled = malonSingingDisabled;
 
-    D_8016B9F2 = arg0;
     if ((Audio_GetActiveSeqId(SEQ_PLAYER_BGM_MAIN) & 0xFF) == NA_BGM_LONLON) {
+        // Malon singing in Lon Lon Ranch
         seqPlayerIndex = SEQ_PLAYER_BGM_MAIN;
```

Comment:
there are comments here like "singing in lon lon ranch" and "in night market", but from a surface level it doesnt seem the audio system is aware of that/checking based on scene or anything. Seems like we may just be making assumptions based on how it ends up getting used in practice. Feel like it probably makes sense for this to be a bit more generalized? but I may be missing something that makes them more tied to location

Reply:
reading it again it seems like the only distinction the code has here is "current sequence is lon lon bgm" and "current sequence is lon lon bgm with ambience also playing". So yea I think it would be good to remove the location assumptions from the comments